src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfig.java

```diff
@@ -0,0 +1,269 @@
+          + "Each record is composed of columns (also called fields). "
+          + "Every table is defined by a schema that describes the column names, data types, and other information.")
+  private String table;
+
+  @Name(NAME_ARGUMENT_SELECTION_CONDITIONS)
+  @Macro
+  @Description(
+      "A set of conditions for identifying the arguments to run a pipeline."
+          + "A particular use case for this would be feed=marketing;date=20200427.")
+  private String argumentSelectionConditions;
+
+  @Name(NAME_ARGUMENTS_COLUMNS)
+  @Macro
+  @Description(
+      "Name of the columns, separated by `,` ,that contains the arguments for this run."
+          + "A particular use case for this would be country, device")
+  private String argumentsColumns;
+
+  public BigQueryArgumentSetterConfig(
+      String referenceName,
+      String dataset,
+      String table,
+      String argumentSelectionConditions,
+      String argumentsColumns) {
+    this.referenceName = referenceName;
+    this.dataset = dataset;
+    this.table = table;
+    this.argumentSelectionConditions = argumentSelectionConditions;
+    this.argumentsColumns = argumentsColumns;
+  }
+
+  public String getDataset() {
+    return dataset;
+  }
+
+  public String getTable() {
+    return table;
+  }
+
+  public String getArgumentSelectionConditions() {
+    return argumentSelectionConditions;
+  }
+
+  @Nullable
+  public String getArgumentsColumns() {
+    return argumentsColumns;
+  }
+
+  @Override
+  public void validate(FailureCollector collector) {
+    super.validate(collector);
+    validateProperties(collector);
+
+    if (canConnect()) {
+      try {
+        getQueryJobConfiguration();
+      } catch (Exception e) {
+        collector.addFailure(e.getMessage(), "");
+      }
+    }
+  }
+
+  public void validateProperties(FailureCollector collector) {
+    if (!containsMacro(NAME_DATASET)) {
+      BigQueryUtil.validateDataset(dataset, NAME_DATASET, collector);
+    }
+
+    if (!containsMacro(NAME_TABLE)) {
+      BigQueryUtil.validateTable(table, NAME_TABLE, collector);
+    }
+
+    validateArgumentsSelectionConditions(getArgumentSelectionConditions(), collector);
+    validateArgumentsColumns(argumentsColumns, collector);
+  }
+
+  private boolean canConnect() {
+    return !containsMacro(NAME_DATASET)
+        && !containsMacro(NAME_TABLE)
+        && !containsMacro(NAME_DATASET_PROJECT)
+        && !containsMacro(NAME_SERVICE_ACCOUNT_FILE_PATH)
+        && !containsMacro(NAME_PROJECT)
+        && !containsMacro(NAME_ARGUMENT_SELECTION_CONDITIONS)
+        && !containsMacro(argumentsColumns);
+  }
+
+  private void validateArgumentsSelectionConditions(
+      String argumentSelectionConditions, FailureCollector collector) {
+    if (containsMacro(NAME_ARGUMENT_SELECTION_CONDITIONS)) {
+      return;
+    }
+
+    if (Strings.isNullOrEmpty(argumentSelectionConditions)) {
+      collector.addFailure(
+          "Arguments Selection Conditions is empty!",
```

Comment:
nit: `empty.`
Please add period to end of failure message and corrective action